### PR TITLE
Refactor game logic, infect the entire project with TypeScript

### PR DESCRIPTION
**Pull request by nmoadev**
* Wrangled babel so that TS files don't need to be imported by extension, which makes linter cry
* Refactord the game log heavily so that projects just convert one tile to another
* Added "play_game_logic.ts" which actually plays a few turns and  you can see a tile change from one building to another in the console.
* Added lots of documentation to make the log more understandable

### CODE REVIEW

File: src/play_game_logic.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { advanceTurn, createGameState, playerAssignWorkers, playerInitiateProject, TileUnderConstruction } from "./game_logic/index";
```
Comment (nmoadev) [inline src/play_game_logic.ts:1]: @scottysseus / @libjared This file actually shows a "game" being played

File: src/game_logic/interfaces.ts [hunk 2]
```diff
@@ -1,59 +1,98 @@
...
+  readonly description: string
+  /**
+   * Look at the current game state and determine if this contract has been fulfilled
+   * 
+   * changes `isCompeleted()` if has been completed
+   * 
+   * DO NOT modify the game state
+   * @param state 
+   */
   check(state: GameState): boolean
```
Comment (scottysseus) [inline src/game_logic/interfaces.ts:80]: I do like the flexibility that passing the whole game state affords

**Review by scottysseus — COMMENTED** [review]
I am a little worried about the move from `.js` to `.tsx` - do we need to start adding types to our React code?

**Comment by hypevhs** [comment]
> I am a little worried about the move from `.js` to `.tsx` - do we need to start adding types to our React code?

I think TS is a superset of JS so you don't have to "convert" existing code if you don't want to. All TS syntax is optional. Apparently.

**Pull request merged**